wenet/dataset/processor.py

```diff
@@ -292,13 +339,20 @@ def tokenize(data, symbol_table, bpe_model=None, split_with_space=False):
         Returns:
             Iterable[{key, wav, txt, tokens, label, sample_rate}]
     """
+    if non_lang_syms is not None:
+        rs = [re.compile(re.escape(x)) for x in non_lang_syms]
+    else:
+        rs = []
+
     if bpe_model is not None:
         import sentencepiece as spm
         sp = spm.SentencePieceProcessor()
         sp.load(bpe_model)
+
     for sample in data:
         assert 'txt' in sample
         txt = sample['txt'].strip()
+        txt = __tokenize_non_lang_syms(txt, rs)
```

Comment:
``` python
if len(rs) > 0:
    __tokenize_non_lang_syms(txt, rs)
```

Reply:
This function just returns input txt if rs is an empty list

Reply:
Please also take bpe mode into consideration into account. The code fails for BPE.

Reply:
> 

we should avoid to call the function to save time especially for very big data.